src/dndbeyond/content-scripts/character.js

```diff
@@ -857,24 +857,97 @@ function displayAction(paneClass) {
     });
 }
 
+function handleCustomText(paneClass) {
+    // Function Objective: Search for and handle block of VTT Message text...
+    //...in pane Notes or Description (Priority descending in that order.)
+    const regexp = /```\[([^\]]*)\]\s*((\S|\s)*?)\s*```/g;
+    const rollOrderTypes = ["before", "after", "replace"];//(relative to normal roll msg)
+    const defaultRollFunc = function () {};
+    let rollOrder = rollOrderTypes[0];
+    let rollFunc = defaultRollFunc;
+    
+    let pane = $("."+paneClass);
+    let elemText;
+    let matchedText;
+    
+    // Search for msg text in "Notes" section, if it exists
+    //Notes: Iterate through classes in pane: "ddbc-property-list__property" that contain the text "Notes:"
+    //  If exists: load text from "ddbc-property-list__property-content" (Last child of "ddbc-property-list__property")
+    let msgElement = pane.find(".ddbc-property-list__property:contains('Notes:')").get(0)
+    if (msgElement!=undefined && msgElement!=null) {
+        elemText = msgElement.lastChild.innerText;
+        // console.log("Notes Text: "+elemText);
+        matchedText = [...elemText.matchAll(regexp)];
+    }
+
+    // If no message was found, search for msg in the "Description" section, if it exists
+    if (matchedText==undefined || matchedText==null) {
+        msgElement = pane.find(".ct-action-detail__description").get(0);
+        console.log(msgElement);
+        if (msgElement!=undefined) {
+            elemText = msgElement.innerText;
+            // console.log("Description Text: "+elemText);    
+            matchedText = [...elemText.matchAll(regexp)]; 
+        }
+    }
+
+    //TODO possibly add the ability to have the note text add its own, separate roll button, instead of using the existing ones
+    //Handle Matched Text
+    if (matchedText!=undefined && matchedText!=null) {
+        console.log("Matched Text:\n"+matchedText);
+        for (let i=0; i<matchedText.length; i++) {
+            let cMatchType = matchedText[i][1].toLowerCase();
+            let cMatchText = matchedText[i][2];
+            // Determine if we should prevent the default action/message from being sent to the VTT
+            console.log("Note Match #"+i+" (Provided Type: "+cMatchType+"):\n"+cMatchText);
+            if (rollOrderTypes.includes(cMatchType))
+                rollOrder = cMatchType;
+            rollFunc = function () { 
+                sendRollWithCharacter("action", 0, {
+                    "type": "raw-text",
+                    "text": cMatchText
+                })
+            };
+        }
+    }
```

Comment:
This block doesn't break when it finds one match, so if you have multiple, it will use the last but might use the roll order from another line if the last one didn't match a valid order type, Also the rollFunc gets set and rollOrder remain as 'before' if you simply have `[]`.
The modified regexp which checks for specific valid values, and the use of `match` instead of `matchAll` should fix those issues. Here, I would just replace it with : 
```js
if (customMatch) {
  const sendChat = function () { 
                sendRollWithCharacter("action", 0, {
                    "type": "chat-message",
                    "text": customMatch[2]
                })
            };
  switch(customMatch[1]) {
   case "before-beyond20":
      sendChat();
      executeRoll();
   break;
   case "after-beyond20":
      executeRoll();
      sendChat();
   break;
   case "replace-beyond20":
      sendChat();
   break;
  default:
   // Shouldn't happen, but just in case, fallback
  executeRoll()
   break;
  }
} else {
  executeRoll();
}
```